### ISSUE TITLE
Reroute edges for organizing nodes

### ISSUE THREAD

**Issue by Lusbert** [chaiNNer-org/chaiNNer#2420]
**Motivation**
I often have noodles from nodes crossing other nodes and it makes it hard to organize things and often I lose track of which noodle is connected to which node.

**Description**
Just like in **Blender** with the *Reroute* function.
## without Reroute
![noReroute](https://github.com/chaiNNer-org/chaiNNer/assets/118057562/2778c34b-5a2d-4a22-92f8-bf95cc83675e)

## with Reroute
with reroute I can simply *route* my nodes for better organization
![Reroute](https://github.com/chaiNNer-org/chaiNNer/assets/118057562/ac9670e9-c707-48c3-8918-ef73a48a51c3)

**Alternatives**
some kind of node that can be used as a portal, for example:
I input a node into *Portal A* and i can get the node from another *Portal A* at the location i need my node.
Basically like the game Portal

**Comment by RunDevelopment** [comment]
Great suggestion! Implementing this might be a bit difficult, but we'll probably add it eventually.

Those "noodles" are called "links" or "edges" btw :)

> some kind of node that can be used as a portal, for example:

Already suggested in #833.

**Comment by joeyballentine** [comment]
I've wanted to do something like that for a whilw. The tricky part is making it work with our "hold alt and drag a node onto an edge to put it in between" feature. Maybe we could functionally implement each spot as a special type of node and therefore each part of the edge would be a separate edge, and each dot would be a passthrough? Something like that might work

**Comment by Lusbert** [comment]
> Those "noodles" are called "links" or "edges" btw :)

lol thanks i didn't know that :0
